.github/workflows/main.yml

```
@@ -92,3 +92,43 @@ jobs:
           python -m pip install --upgrade pip setuptools tox
       - name: Test with tox
         run: tox -e py${{ matrix.tox-py-version }}-pytest${{ matrix.pytest-version }}
+
+  mypy-run:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11-dev" ]
+        include:
+          # Convert python-version to a string that tox understands
+          - python-version: "3.7"
+            tox-py-version: "37"
+            ignore-mypy-error: "true"
+          - python-version: "3.8"
+            tox-py-version: "38"
+            ignore-mypy-error: "true"
+          - python-version: "3.9"
+            tox-py-version: "39"
+            ignore-mypy-error: "true"
+          - python-version: "3.10"
+            tox-py-version: "310"
+            ignore-mypy-error: "false"
+          - python-version: "3.11-dev"
+            tox-py-version: "311"
+            ignore-mypy-error: "false"
```

NIP: here we probably want to set `ignore-mypy-error` to `true`, because `mypy` is run against unstable/dev python version:

```suggestion
            ignore-mypy-error: "true"
```

---

it's now in beta, so it's stable enough for me :). By the time the first failure happens in 3.11, it will be probably be a stable release of 3.11